emmet-core/emmet/core/mpid.py

```diff
@@ -18,8 +26,8 @@ def __init__(self, val: Union["MPID", int, str]):
             self.parts = val.parts  # type: ignore
             self.string = val.string  # type: ignore
 
-        elif isinstance(val, int):
-            self.parts = (None, val)
+        elif isinstance(val, int) or isinstance(val, str) and val.isnumeric():
```

Comment:
Could you wrap this in parentheses? Just for readability to distinguish `(A or B) and C` vs `A or (B and C)`